src/test/java/com/arpnetworking/metrics/common/sources/CollectdHttpSourceTest.java

```diff
@@ -0,0 +1,172 @@
+                .utf8String();
+        Assert.assertEquals(400, response.status().intValue());
+        Assert.assertTrue(errorString.contains("service cannot be null"));
+        Assert.assertTrue(errorString.contains("cluster cannot be null"));
+    }
+
+    @Test
+    public void testParsesTagHeaders() throws ParsingException, ExecutionException, InterruptedException {
+        final DefaultRecord.Builder builder = TestBeanFactory.createRecordBuilder();
+        builder.setService(null);
+        builder.setCluster(null);
+
+        final String service = "my-service";
+        final String cluster = "my-cluster";
+        final HttpRequest request = HttpRequest.create()
+                .addHeader(RawHeader.create("X-TAG-SERVICE", service))
+                .addHeader(RawHeader.create("X-TAG-CLUSTER", cluster));
+        Mockito.when(_parser.parse(Mockito.any())).thenReturn(Collections.singletonList(builder));
+        final HttpResponse response = dispatchRequest(request);
+        Assert.assertEquals(200, response.status().intValue());
+
+        final Record record = builder.build();
+        Assert.assertEquals(cluster, record.getCluster());
+        Assert.assertEquals(service, record.getService());
+    }
+
+    @Test
+    public void testSendsRecords() throws ParsingException, ExecutionException, InterruptedException {
+        final DefaultRecord.Builder builder1 = TestBeanFactory.createRecordBuilder();
+        final DefaultRecord.Builder builder2 = TestBeanFactory.createRecordBuilder();
+        final DefaultRecord.Builder builder3 = TestBeanFactory.createRecordBuilder();
+        final ArrayList<DefaultRecord.Builder> builders = Lists.newArrayList(builder1, builder2, builder3);
+
+        Mockito.when(_parser.parse(Mockito.any())).thenReturn(builders);
+        final HttpResponse response = dispatchRequest();
+        Assert.assertEquals(200, response.status().intValue());
+        final ArgumentCaptor<Record> builderCaptor = ArgumentCaptor.forClass(Record.class);
+        Mockito.verify(_observer, Mockito.times(3)).notify(Mockito.any(), builderCaptor.capture());
+
+        final List<Record> constructed = builderCaptor.getAllValues();
+        Assert.assertEquals(builder1.build(), constructed.get(0));
+        Assert.assertEquals(builder2.build(), constructed.get(1));
+        Assert.assertEquals(builder3.build(), constructed.get(2));
+    }
+
+    private HttpResponse dispatchRequest() throws ExecutionException {
+        return dispatchRequest(HttpRequest.create());
+    }
+
+    private HttpResponse dispatchRequest(final HttpRequest request) {
+        final ActorRef ref = getSystem().actorOf(CollectdHttpSource.Actor.props(_source));
+        final CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+        final RequestReply message = new RequestReply(request, future);
+
+        ref.tell(message, ActorRef.noSender());
+
+        try {
+            return future.get(10000, TimeUnit.MILLISECONDS);
```

Comment:
That's a timeout, not a sleep.  If your tests *actually took 5 seconds* it's not too long.  This will not sleep for 10 seconds if it takes 1/10 of a second to complete.